hooks/queries/useMapRoute.tsx

```diff
@@ -1,31 +1,30 @@
 import { useQuery } from "@tanstack/react-query";
 import axios from "axios";
+import { useEffect, useState } from "react";
 
 const mapboxToken = process.env.NEXT_PUBLIC_MAP_BOX_TOKEN;
 
 export const useMapRoute = ({
-  venuesPlan,
+  venues,
   startPoint,
   endPoint,
 }: {
-  venuesPlan: Venue[];
+  venues: Venue[] | undefined;
   startPoint: MapLocation;
   endPoint: MapLocation;
 }) => {
-  const { data: routeData } = useQuery(
-    ["points", venuesPlan, startPoint, endPoint],
-    () => {
-      const step = "20";
-      const radiuses = venuesPlan.map(() => step);
-      if (startPoint.place_name != "") radiuses.push(step);
-      if (endPoint.place_name != "") radiuses.push(step);
-      const radiusesString = radiuses.join(";");
-
-      return axios.get(
-        `https://api.mapbox.com/matching/v5/mapbox/driving/${routePoints}?geometries=geojson&radiuses=${radiusesString}&access_token=${mapboxToken}`
+  const [routePoints, setRoutePoints] = useState("");
+
+  useEffect(() => {
+    if (venues) {
+      const points = getPlanPoints(
```

Comment:
Not sure whats going on here but doesn't feel right?

Reply:
Agree this function name isn't great, maybe it should be something like getPlanPointsAsQueryString(). I also think I can use axios parameter serialisation here instead.